pkg/occlient/occlient.go

```diff
@@ -681,6 +718,102 @@ func secretKeyName(componentName, baseKeyName string) string {
 	return fmt.Sprintf("COMPONENT_%v_%v", strings.Replace(strings.ToUpper(componentName), "-", "_", -1), strings.ToUpper(baseKeyName))
 }
 
+// getS2ILabelValue returns the requested S2I label value from the passed set of labels attached to builder image
+// and the hard coded possible list(the labels are not uniform across different builder images) of expected labels
+func getS2ILabelValue(labels map[string]string, expectedLabelsSet []string) string {
+	for _, label := range expectedLabelsSet {
+		if retVal, ok := labels[label]; ok {
+			return retVal
+		}
+	}
+	return ""
+}
+
+// GetS2IPathsFromBuilderImg returns script path protocol, S2I scripts path, S2I source or binary expected path, S2I deployment dir and errors(if any) from the passed builder image
+func GetS2IPathsFromBuilderImg(builderImage *imagev1.ImageStreamImage) (string, string, string, string, error) {
```

Comment:
It would be better to remove struct than 5 independent strings. It is hard to know what value is what.

Reply:
Ack
Done